src/main/kotlin/io/github/mojira/arisa/infrastructure/ProjectCache.kt

```diff
@@ -0,0 +1,40 @@
+package io.github.mojira.arisa.infrastructure
+
+import io.github.mojira.arisa.jiraClient
+import java.time.Duration
+import java.time.Instant
+import net.rcarz.jiraclient.Project as JiraProject
+
+object ProjectCache {
+    private var cache = Cache<JiraProject>()
+    private var lastRefresh = Instant.now()
+
+    private const val REFRESH_INTERVAL_IN_MINUTES: Long = 5
+
+    fun getProjectFromTicketId(id: String): JiraProject {
+        checkForRefresh()
+
+        val projectKey = id.split("-")[0]
```

Comment:
This custom parsing logic might not be necessary, rcarz's `Issue` has a `getProject()` method from which we could get the project key.

Reply:
Yeah, I noticed this as well after this was merged... but I figured it doesn't really matter since in the end it does the same thing.